grails-app/controllers/au/org/ala/volunteer/NewsItemController.groovy

```diff
@@ -0,0 +1,278 @@
+    }
+
+    @Transactional
+    def toggleNewsItemStatus(NewsItem newsItem) {
+        log.debug("Toggling status for news item: ${newsItem?.id}, current status: ${newsItem?.isActive}")
+        if (!newsItem) {
+            render status: 404
+            return
+        }
+        if (!userService.isSiteAdmin()) {
+            log.debug("No permission to toggle news item status, user is not a site admin.")
+            response.sendError(SC_FORBIDDEN, "You don't have permission")
+            return
+        }
+        log.debug("verifyId: ${params.verifyId}, newsItem.id: ${newsItem.id}, params.verifyId as long == newsItem.id: ${params.verifyId as long == newsItem.id}")
+        log.debug("verify check: ${!params.verifyId} || ${params.verifyId as long != newsItem.id}")
+        if (params.verifyId == null || params.verifyId as long != newsItem.id) {
+            log.debug("Verification ID does not match news item ID, user does not have permission to toggle status.")
+            flash.message = "You do not have permission to view this page"
+            render(view: '/notPermitted')
+            return
+        }
+
+        log.debug("Toggling status for news item: ${newsItem.id}, current status: ${newsItem.isActive}")
+        newsItem.isActive = (!newsItem.isActive)
+        log.debug("New status for news item: ${newsItem.isActive}")
+        newsItem.save(flush: true, failOnError: true)
+        flash.message = "The news item's status has been set to ${newsItem.isActive ? 'active' : 'inactive'}."
+        redirect(uri: request?.getHeader("referer") ?: createLink(controller: 'newsItem', action: 'manage'))
+    }
+
+    def create() {
+        if (!userService.isSiteAdmin()) {
+            flash.message = "You do not have permission to view this page"
+            render(view: '/notPermitted')
+            return
+        }
+        def datePickerRanges = getDatePickerRanges()
+        log.debug("End date: ${datePickerRanges.endDate}")
+        log.debug("Start date: ${datePickerRanges.startDate}")
+        render view: 'create', model: [defaultEndDate: datePickerRanges.endDate, defaultStartDate: datePickerRanges.startDate]
+    }
+
+    /**
+     * Returns the date picker ranges for the news item creation form.
+     * @return a map containing the start and end date strings formatted according to the specified pattern.
+     */
+    private def getDatePickerRanges() {
+        String endDateStr = LocalDate.now().plusDays(30).format(DateTimeFormatter.ofPattern(DateConstants.DATE_FORMAT_SHORT, Locale.ENGLISH))
+        String startDateStr = LocalDate.now().format(DateTimeFormatter.ofPattern(DateConstants.DATE_FORMAT_SHORT, Locale.ENGLISH))
+        return [startDate: startDateStr, endDate: endDateStr]
+    }
+
+    @Transactional
+    def save() {
+        log.debug("Saving news item with params: ${params}")
+        if (!userService.isSiteAdmin()) {
```

Comment:
You could probably pull this role check into an annotation or interceptor if you felt like saving a few lines of code per method.